monitor.js

```diff
@@ -33,123 +47,183 @@ class TimedCounter extends Map {
 // 5 = supposedly completely silent, but for now a lot of PS output doesn't respect loglevel
 if (Config.loglevel === undefined) Config.loglevel = 2;
 
-const Monitor = module.exports = {
-
+module.exports = {
 	/*********************************************************
 	 * Logging
 	 *********************************************************/
 
-	log: function (text) {
+	/**
+	 * @param {string} text
+	 */
+	log(text) {
 		this.notice(text);
 		if (Rooms('staff')) {
-			Rooms('staff').add('|c|~|' + text).update();
+			Rooms('staff').add(`|c|~|${text}`).update();
 		}
 	},
-	adminlog: function (text) {
+
+	/**
+	 * @param {string} text
+	 */
+	adminlog(text) {
 		this.notice(text);
 		if (Rooms('upperstaff')) {
-			Rooms('upperstaff').add('|c|~|' + text).update();
+			Rooms('upperstaff').add(`|c|~|${text}`).update();
 		}
 	},
-	logHTML: function (text) {
+
+	/**
+	 * @param {string} text
+	 */
+	logHTML(text) {
 		this.notice(text);
 		if (Rooms('staff')) {
-			Rooms('staff').add('|html|' + text).update();
+			Rooms('staff').add(`|html|${text}`).update();
 		}
 	},
-	debug: function (text) {
+
+	/**
+	 * @param {string} text
+	 */
+	debug(text) {
 		if (Config.loglevel <= 1) console.log(text);
 	},
-	warn: function (text) {
+
+	/**
+	 * @param {string} text
+	 */
+	warn(text) {
 		if (Config.loglevel <= 3) console.log(text);
 	},
-	notice: function (text) {
+
+	/**
+	 * @param {string} text
+	 */
+	notice(text) {
 		if (Config.loglevel <= 2) console.log(text);
 	},
 
 	/*********************************************************
 	 * Resource Monitor
 	 *********************************************************/
 
-	clean: function () {
-		Monitor.clearNetworkUse();
-		Monitor.battlePreps.clear();
-		Monitor.battles.clear();
-		Monitor.connections.clear();
+	clean() {
+		this.clearNetworkUse();
+		this.battlePreps.clear();
+		this.battles.clear();
+		this.connections.clear();
 		Dnsbl.cache.clear();
 	},
+	// @ts-ignore
+	cleanInterval: setInterval(() => this.clean(), 2 * 60 * 60 * 1000),
```

Comment:
Shit, it doesn't. I thought `this` would point to the `module.exports` object regardless of where in the object `this` is referred to, rather than just in methods directly on the object. I'll change it back to how it was defined before.